src/puppet.ts

```diff
@@ -257,19 +257,19 @@ export abstract class Puppet extends EventEmitter {
    *
    *
    */
-  public emit (event: 'dong',         data?: string)                                                                                    : boolean
-  public emit (event: 'error',        error: Error)                                                                                     : boolean
-  public emit (event: 'friendship',   friendshipId: string)                                                                             : boolean
-  public emit (event: 'login',        contactId: string)                                                                                : boolean
-  public emit (event: 'logout',       contactId: string)                                                                                : boolean
-  public emit (event: 'message',      messageId: string)                                                                                : boolean
-  public emit (event: 'reset',        reason: string)                                                                                   : boolean
-  public emit (event: 'room-join',    roomId: string, timestamp: number, inviteeIdList:  string[], inviterId: string)                   : boolean
-  public emit (event: 'room-leave',   roomId: string, timestamp: number, leaverIdList:   string[], remover?: string)                    : boolean
-  public emit (event: 'room-topic',   roomId: string, timestamp: number, newTopic:       string,   oldTopic: string, changerId: string) : boolean
-  public emit (event: 'room-invite',  roomInvitationId: string)                                                                         : boolean
-  public emit (event: 'scan',         qrcode: string, status: ScanStatus, data?: string)                                                : boolean
-  public emit (event: 'ready')                                                                                                          : boolean
+  public emit (event: 'dong',         data?: string)                                                                 : boolean
+  public emit (event: 'error',        error: Error)                                                                  : boolean
+  public emit (event: 'friendship',   friendshipId: string)                                                          : boolean
+  public emit (event: 'login',        contactId: string)                                                             : boolean
+  public emit (event: 'logout',       contactId: string)                                                             : boolean
+  public emit (event: 'message',      messageId: string)                                                             : boolean
+  public emit (event: 'reset',        reason: string)                                                                : boolean
+  public emit (event: 'room-join',    roomId: string, inviteeIdList:  string[], inviterId: string)                   : boolean
```

Comment:
How can you get to know the timestamp for the listeners, when you do not have a timestamp argument for emit?